code/game/objects/structures/extinguisher.dm

```diff
@@ -54,6 +53,8 @@
 		return
 	if(istype(O, /obj/item/extinguisher))
 		if(!has_extinguisher && opened)
+			if((O.flags & NODROP) || (flags & NODROP))
```

Comment:
This isn't how you do it

`If(!user.drop_item() return`

Reply:
Changed to this solution.